docs/reference-guide/metadata.md

```diff
@@ -0,0 +1,88 @@
+---
+id: metadata
+title: Metadata
+sidebar_label: Metadata
+---
+
+Bolt protocol specifies additional data that can be sent along with the
+requested results. Such data is called metadata and can be divided into two
+groups:
+ - Query Statistics,
+ - Notifications.
+
+Both of these metadata can be accessed through `summary` map that is being sent
+along with the results of the query. Query statistics will be under `stats` key,
+and notifications under `notifications` key.
+
+## Query Statistics
+
+Query statistics will be sent whenever a user performs a query that will effect
+data in any way. In other words we will track the quantity of these changes
+throughout the query execution and report it back to the user.
+
+The structure of statistics is a map of string keys and integer values. Data
+that is being tracked:
+
+ - `nodes-created`
+ - `nodes-deleted`
+ - `relationships-created`
+ - `relationships-deleted`
+ - `labels-added`
+ - `labels-removed`
+ - `properties-set`
+
+This data will refer only to the changes done by query, not triggers changes will
+not affect these values.
```

Comment:
This is wrong actually because `BEFORE COMMIT` will run trigger statements as part of the transaction so it should affect stats.

Reply:
Hmm it didn't affected the stats

Reply:
I've tested both BEFORE and AFTER commit neither of these shows up in stats, only the query changes.